### PR TITLE
Add panic topic to push exceptions to Intercom.

### PR DESCRIPTION
**Pull request by jasebell**
Branch development PR: 

Added topic to push panic exceptions through to Intercom for inspection. For example in the local testing 

`{:topic :activity/panic, :args {:message "something", :activity :upload-file, :activity-name :upload-file, :id "edb1fb66-0945-4dd7-b9b4-01d2300e4695"}}`

Current WIP commit is for the basic structure of the panic alert and some refactoring of the original intercom.cljs code. The function `abandon-activity!` in activities.cljs needs to pull a reporter for more info. This still needs fixing.

### CODE REVIEW

File: src/cljs/witan/ui/activities.cljs [hunk 1]
```diff
@@ -169,7 +169,13 @@
   (let [{:keys [activity] :as app-state}
         (data/get-in-app-state :app/activities :activities/pending id)]
     (when (and app-state (= activity-name activity))
-      (finish-activity! nil :failed id))))
+      ;; FIXME - Change message to something useful from a reporter.
+      (do
+        (data/publish-topic :activity/panic {:message "something"
```
Comment (otfrom) [inline src/cljs/witan/ui/activities.cljs]: I'm in two minds about whether or not we need this as the `finish-activity!` does catch the error (but not as clearly as the panic will). I wonder if they are 2 different things.

File: src/cljs/witan/ui/data.cljs [hunk 2]
```diff
@@ -141,6 +141,8 @@
 (defn panic!
   [msg]
   (log/severe "App has panicked:" msg)
+  ;; FIXME HERE
```
Comment (otfrom) [inline src/cljs/witan/ui/data.cljs]: This `panic!` function needs to move below the `publish-topic` function below.

File: src/cljs/witan/ui/controllers/intercom.cljs [hunk 3]
```diff
@@ -21,27 +21,39 @@
...
+(defn publish-to-intercom [event-label name activity payload message]
+  (.Intercom js/window
+             event-label
+             (name activity)
+             (clj->js (parse-payload payload message))))
+
+(defn on-panic-event
+  [{:keys [args]}]
+  (let [{:keys [activity message] :as payload} args]
+    (publish-to-intercom "panicEvent" name activity payload message)))
```
Comment (otfrom) [inline src/cljs/witan/ui/controllers/intercom.cljs]: I don't understand the difference between panicEvent and trackEvent, but maybe @acron0 can explain.

**Review by acron0 — APPROVED** [review]
Changes were discussed and made

**Pull request merged**